### PR TITLE
feat: add maxLength option

### PR DESCRIPTION
**Pull request by AndrewLeedham**
fixes: #26 

I feel leaving the 2900 limit as default is sensible, because most people will not need a massive limit, but the option allows them to increase it if necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,14 +80,15 @@ Alternatively, you could pass the webhook as a configuration option.
...
+| `onSuccessTemplate`    | Provides a template for the slack message object on success when `notifyOnSuccess` is `true`. See [templating](#templating).                                              | undefined     |
+| `onFailTemplate`       | Provides a template for the slack message object on fail when `notifyOnFail` is `true`. See [templating](#templating).                                                    | undefined     |
+| `markdownReleaseNotes` | Pass release notes through markdown to slack formatter before rendering.                                                                                                  | false         |
+| `slackWebhook`         | Slack webhook created when adding app to workspace.                                                                                                                       | SLACK_WEBHOOK |
+| `maxLength`            | Maximum character length for the release notes before truncation. [Slacks hard limit is 40,000](https://api.slack.com/changelog/2018-04-truncating-really-long-messages). | 2900          |
```
Comment (juliuscc) [inline README.md]: As it is not as simple as a 40000 character limit, do you think it makes more sense to say something like:

> Maximum character length for the release notes before truncation. If `maxLength` is too high, messages can be dropped. [Read here](https://github.com/juliuscc/semantic-release-slack-bot/issues/26#issuecomment-569804359) for more information.

File: lib/success.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const template = require('./template')
 const truncate = require('./truncate')
 const getRepoInfo = require('./getRepoInfo')
 
-// Max lenght of message to the slack api is 3000 characters. Make sure we are below that.
+// Max length of message to the slack api is 40,000 characters. 2900 is based on the old limit.
```
Comment (juliuscc) [inline lib/success.js]: I think the old comment is wrong, as the limit is specific to `section`. The new comment does not really explain why `MAX_LENGTH ` is 2900 though. Do you think we should change the last sentence to something that explains why 2900 is chosen?

Like `2900 is the limit for a message block of type 'section'.`

File: lib/success.js [hunk 3]
```diff
@@ -15,7 +15,10 @@ module.exports = async (pluginConfig, context) => {
 		options,
 		env: { SEMANTIC_RELEASE_PACKAGE, npm_package_name }
 	} = context
-	const { slackWebhook = process.env.SLACK_WEBHOOK } = pluginConfig
+	const {
+		slackWebhook = process.env.SLACK_WEBHOOK,
+		maxLength = MAX_LENGTH
```
Comment (juliuscc) [inline lib/success.js]: Let's change `maxLength` to something with 'unsafe' in it.This change also has to happen in README.md

File: README.md [hunk 4]
```diff
@@ -29,17 +29,17 @@ The plugin can be configured in the [**semantic-release** configuration file](ht
...
+		"@semantic-release/commit-analyzer",
+		"@semantic-release/release-notes-generator",
+		[
+			"semantic-release-slack-bot",
+			{
+				"notifyOnSuccess": false,
+				"notifyOnFail": true
+			}
+		]
+	]
```
Comment (juliuscc) [inline README.md]: I think an 8 space indentation looks strange in a markdown file. Semantic-release uses 2 spaces in their documentation so I think we should try to align to them.

Comment (AndrewLeedham) [reply README.md]: Done

File: lib/success.js (hunk 3 above)
Comment (AndrewLeedham) [reply lib/success.js]: Done

File: lib/success.js (hunk 2 above)
Comment (AndrewLeedham) [reply lib/success.js]: Done

File: README.md (hunk 1 above)
Comment (AndrewLeedham) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (juliuscc) [reply README.md]: Now there are tabs. I looked at the prettier config and it specifies tabs, but we should use 2 spaces. That is another issue that I will look into.

Could you remove this chunk from the PR, and I will update the prettier config in another PR.

**Comment by AndrewLeedham** [comment]
> I only have one small comment but otherwise the PR looks good! Thanks!

Yup I did revert that but I guess when I saved the new options table it formatted it again 🤦‍♂

**Comment by juliuscc** [comment]
It is still there but I think it is because of husky, and I can fix this when I update the prettier config. Thanks for your PR!

**Pull request merged**